src/ObjCRuntime/Registrar.cs

```diff
@@ -186,7 +186,7 @@ public void VerifyRegisterAttribute (ref List<Exception> exceptions)
 				}
 			}
 
-			static char[] invalidSelectorCharacters = new char[] { ' ', '\t' };
+			static char[] invalidSelectorCharacters = new char[] { ' ', '\t', '?', '\\', '!', '|', '@', '"', '\'', '%', '&', '/', '(', ')', '=', '^', '[', ']', '{', '}', ',', '.', ';', '-', '\n' };
```

Comment:
If we don't do this, at least consider adding a comment pointing to the other duplicate in the test.